cloudevents/pydantic/__init__.py

```diff
@@ -11,7 +11,29 @@
 #    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 #    License for the specific language governing permissions and limitations
 #    under the License.
-from cloudevents.pydantic.conversion import from_dict, from_http, from_json
-from cloudevents.pydantic.event import CloudEvent
+from cloudevents.exceptions import PydanticFeatureNotInstalled
+
+try:
+    from pydantic import VERSION as PYDANTIC_VERSION
+
+    pydantic_major_version = PYDANTIC_VERSION.split(".")[0]
+    if pydantic_major_version == "2":
+        from cloudevents.pydantic.v1.conversion import from_dict, from_http, from_json
+        from cloudevents.pydantic.v1.event import CloudEvent
+
+    else:
+        from cloudevents.pydantic.v2.conversion import (  # type: ignore
```

Comment:
Same typing error as previous pull request due to re-assignment of different objects 

https://github.com/cloudevents/sdk-python/pull/218/files#diff-3feb86f607bffb91257e35096edfa40b2d6a291cdc66d404dc139c71531e8b1bR27